pkg/cmd/cli/cmd/set/buildsecret.go

```diff
@@ -0,0 +1,283 @@
+	}
+
+	cmdNamespace, explicit, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	clientConfig, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+	o.OutputVersion, err = kcmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
+	if err != nil {
+		return err
+	}
+
+	mapper, typer := f.Object(false)
+	o.Builder = resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(explicit, false, o.Filenames...).
+		SelectorParam(o.Selector).
+		ResourceNames("buildconfigs", resources...).
+		Flatten()
+
+	if o.All {
+		o.Builder.ResourceTypes("buildconfigs").SelectAllParam(o.All)
+	}
+
+	output := kcmdutil.GetFlagString(cmd, "output")
+	if len(output) != 0 {
+		o.PrintObject = func(obj runtime.Object) error { return f.PrintObject(cmd, mapper, obj, o.Out) }
+	}
+
+	o.Encoder = f.JSONEncoder()
+	o.ShortOutput = kcmdutil.GetFlagString(cmd, "output") == "name"
+	o.Mapper = mapper
+
+	return nil
+}
+
+func (o *BuildSecretOptions) Validate() error {
+	if !o.Pull && !o.Push && !o.Source {
+		return fmt.Errorf("specify the type of secret to set (--push, --pull, or --source)")
+	}
+	if !o.Remove && len(o.Secret) == 0 {
+		return fmt.Errorf("specify a secret to set")
+	}
+	return nil
+}
+
+func (o *BuildSecretOptions) Run() error {
+	infos := o.Infos
+	singular := len(o.Infos) <= 1
+	if o.Builder != nil {
+		loaded, err := o.Builder.Do().IntoSingular(&singular).Infos()
+		if err != nil {
+			return err
+		}
+		infos = loaded
+	}
+
+	patches := CalculatePatches(infos, o.Encoder, func(info *resource.Info) (bool, error) {
+		bc, ok := info.Object.(*buildapi.BuildConfig)
+		if !ok {
+			return false, nil
+		}
+		o.updateBuildConfig(bc)
+		return true, nil
+	})
+
+	if singular && len(patches) == 0 {
+		return fmt.Errorf("%s/%s is not a build config", infos[0].Mapping.Resource, infos[0].Name)
+	}
+
+	if o.PrintObject != nil {
+		object, err := resource.AsVersionedObject(infos, !singular, o.OutputVersion, kapi.Codecs.LegacyCodec(o.OutputVersion))
+		if err != nil {
+			return err
+		}
+		return o.PrintObject(object)
+	}
+
+	failed := false
+	for _, patch := range patches {
+		info := patch.Info
+		if patch.Err != nil {
+			fmt.Fprintf(o.Err, "error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err)
+			continue
+		}
+
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
+			fmt.Fprintf(o.Err, "info: %s %q was not changed\n", info.Mapping.Resource, info.Name)
+			continue
+		}
+
+		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, kapi.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			fmt.Fprintf(o.Err, "error: %v\n", err)
+			failed = true
+			continue
+		}
+
+		info.Refresh(obj, true)
+		kcmdutil.PrintSuccess(o.Mapper, o.ShortOutput, o.Out, info.Mapping.Resource, info.Name, "updated")
+	}
+	if failed {
+		return cmdutil.ErrExit
```

Comment:
Any reason why you are doing this instead of aggregating the errors and returning it aggregated?